KinSDK/KinSDK/Core/KinAccount.swift

```diff
@@ -13,158 +13,22 @@ import KinUtil
-     */
-    func sendTransaction(_ transactionEnvelope: TransactionEnvelope, completion: @escaping SendTransactionCompletion)
-    
-    /**
-     Send a Kin transaction.
-
-     - Parameter transactionEnvelope: The `TransactionEnvelope` to send.
-
-     - Returns: A promise which is signalled with the `TransactionId` or an `Error`.
-     */
-    func sendTransaction(_ transactionEnvelope: TransactionEnvelope) -> Promise<TransactionId>
-
-    /**
-     Retrieve the current Kin balance.
-
-     - Note: The closure is invoked on a background thread.
-     
-     - Parameter completion: A closure to be invoked once the request completes.
-     */
-    func balance(completion: @escaping BalanceCompletion)
-
-    /**
-     Retrieve the current Kin balance.
-
-     - returns: A `Promise` which is signalled with the current balance.
-     */
-    func balance() -> Promise<Kin>
-
-    /**
-     Watch for changes on the account balance.
-
-     - Parameter balance: An optional `Kin` balance that the watcher will be notified of first.
-
-     - Returns: A `BalanceWatch` object that will notify of any balance changes.
-     */
-    func watchBalance(_ balance: Kin?) throws -> BalanceWatch
-
-    /**
-     Watch for changes of account payments.
-
-     - Parameter cursor: An optional `cursor` that specifies the id of the last payment after which the watcher will be notified of the new payments.
-
-     - Returns: A `PaymentWatch` object that will notify of any payment changes.
-    */
-    func watchPayments(cursor: String?) throws -> PaymentWatch
-
-    /**
-     Watch for the creation of an account.
-
-     - Returns: A `Promise` that signals when the account is detected to have the `.created` `AccountStatus`.
-     */
-    func watchCreation() throws -> Promise<Void>
-
-    /**
-     Exports this account as a Key Store JSON string, to be backed up by the user.
-     
-     - parameter passphrase: The passphrase used to create the associated account.
-     - parameter exportPassphrase: A new passphrase, to encrypt the Key Store JSON.
-     
-     - throws: If the passphrase is invalid, or if exporting the associated account fails.
-     
-     - returns: a prettified JSON string of the `account` exported; `nil` if `account` is `nil`.
-     */
-//    func exportKeyStore(passphrase: String, exportPassphrase: String) throws -> String?
-}
-
-final class KinStellarAccount: KinAccount {
+public final class KinAccount {
     internal let stellarAccount: StellarAccount
     fileprivate let node: Stellar.Node
     fileprivate let appId: AppId
 
     var deleted = false
-    
-    var publicAddress: String {
+
+    /**
+     The public address of this account. If the user wants to receive KIN by sending his address
+     manually to someone, or if you want to display the public address, use this property.
+     */
+    public var publicAddress: String {
         return stellarAccount.publicKey!
     }
 
-    var extra: Data? {
+    public var extra: Data? {
```

Comment:
what's that? the extra field of an account?
if that's the case it should be exposed as `AccountData` with `extra` as one of the properties.

Reply:
@natanrolnik did you need this?

Reply:
@cnotethegr8 what about this property? do we still need it as a public?

Reply:
I don't think so.